src/scroll-timeline-css.js

```diff
@@ -99,10 +112,16 @@ export function initCSSPolyfill() {
       const timelineName = parser.getScrollTimelineName(anim.animationName, evt.target);
       if (timelineName) {
         const scrollTimeline = createScrollTimeline(timelineName);
-        if (anim.timeline != scrollTimeline) {
+        const viewTimeline = createViewTimeline(timelineName);
```

Comment:
This section is a bit confusing since a ViewTimeline is a ScrollTimeline. Suggest expanding the createScrollTimeline method to look at the options and then pick which timeline constructor to call based on source or subject being provided.  Downstream of timeline creation it doesn't matter if it is a generic or specialized ScrollTimeline so we can avoid the if/else logic, where both branches are identical in structure, differing only by the timeline variable.